### PR TITLE
Add v2/error_response

### PR DESCRIPTION
**Pull request by jcorbin**

### CODE REVIEW

File: v2/error_response.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    self.code = code;
+    if (id === null || id === undefined) {
+        self.id = Frame.NullId;
+    } else {
+        self.id = id;
+    }
+    self.message = message ? write.bufferOrString(message) : emptyBuffer;
+}
+
+ErrorResponse.TypeCode = 0xff;
```
Comment (kriskowal) [inline v2/error_response.js]: Cute, bookending the name space.

File: v2/error_response.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    read.UInt8,    // code:1
+    read.UInt32BE, // id:4
+    read.buf2      // message~2
+]), function buildErrorRes(results, buffer, offset) {
+    var code = results[0];
+    var id = results[1];
+    var message = results[2];
+    var res = new ErrorResponse(code, id, message);
+    return [null, offset, res];
+});
```
Comment (kriskowal) [inline v2/error_response.js]: I am not sure how much you would want to rely on it, but you could implement this as `read.record({code: read.UInt8, id: read.UInt32BE, message: read.Buffer2})` and get something nicer than numbered fields.

**Comment by kriskowal** [comment]
Looks good to me. :ok:

File: v2/error_response.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    } else {
+        self.id = id;
+    }
+    self.message = message ? write.bufferOrString(message) : emptyBuffer;
+}
+
+ErrorResponse.TypeCode = 0xff;
+
+ErrorResponse.Overhead = 2 + 4 + 1;
+ErrorResponse.MaxMessageSize = Frame.MayBodySize - ErrorResponse.Overhead;
```
Comment (Raynos) [inline v2/error_response.js]: `Frame.Ma_x_BodySize`

Comment (Raynos) [inline v2/error_response.js]: Where is this field used ? Is it for future use ?

**Comment by Raynos** [comment]
lgtm.

**Pull request merged**